### PR TITLE
tests: add integration tests to verify empty diff when protocols are …

### PR DESCRIPTION
**Pull request by harshadixit12**
…reordered in plugins

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -13,7 +13,7 @@ require (
 	github.com/fatih/color v1.18.0
 	github.com/google/go-cmp v0.7.0
 	github.com/kong/go-apiops v0.1.41
-	github.com/kong/go-database-reconciler v1.22.1
+	github.com/kong/go-database-reconciler v1.22.2-0.20250403105251-3a1014c74cf4
```
Comment (harshadixit12) [inline go.mod:16]: I'll bump the version once the failing tests are fixed on main and rebased here.

**Review by Prashansa-K — APPROVED** [review]
